core/src/integration-test/java/com/scalar/db/api/DistributedStorageAdminRepairTableIntegrationTestBase.java

```diff
@@ -0,0 +1,139 @@
+  private static final String COL_NAME10 = "c10";
+  private static final String COL_NAME11 = "c11";
+
+  protected static final TableMetadata TABLE_METADATA =
+      TableMetadata.newBuilder()
+          .addColumn(COL_NAME1, DataType.INT)
+          .addColumn(COL_NAME2, DataType.TEXT)
+          .addColumn(COL_NAME3, DataType.TEXT)
+          .addColumn(COL_NAME4, DataType.INT)
+          .addColumn(COL_NAME5, DataType.INT)
+          .addColumn(COL_NAME6, DataType.TEXT)
+          .addColumn(COL_NAME7, DataType.BIGINT)
+          .addColumn(COL_NAME8, DataType.FLOAT)
+          .addColumn(COL_NAME9, DataType.DOUBLE)
+          .addColumn(COL_NAME10, DataType.BOOLEAN)
+          .addColumn(COL_NAME11, DataType.BLOB)
+          .addPartitionKey(COL_NAME2)
+          .addPartitionKey(COL_NAME1)
+          .addClusteringKey(COL_NAME4, Scan.Ordering.Order.ASC)
+          .addClusteringKey(COL_NAME3, Scan.Ordering.Order.DESC)
+          .addSecondaryIndex(COL_NAME5)
+          .addSecondaryIndex(COL_NAME6)
+          .build();
+
+  protected DistributedStorageAdmin admin;
+
+  protected AdminTestUtils adminTestUtils;
+
+  protected void initialize() throws Exception {}
+
+  protected abstract Properties getProperties();
+
+  protected Properties getStorageProperties() {
+    return getProperties();
+  }
+
+  protected String getNamespace() {
+    return NAMESPACE;
+  }
+
+  protected String getTable() {
+    return TABLE;
+  }
+
+  private void createTable() throws ExecutionException {
+    Map<String, String> options = getCreateOptions();
+    admin.createNamespace(getNamespace(), options);
+    admin.createTable(getNamespace(), getTable(), TABLE_METADATA, options);
+  }
+
+  protected Map<String, String> getCreateOptions() {
+    return Collections.emptyMap();
+  }
+
+  private void dropTable() throws ExecutionException {
+    admin.dropTable(getNamespace(), getTable());
+    admin.dropNamespace(getNamespace());
+  }
+
+  @BeforeAll
+  protected void beforeAll() throws Exception {}
+
+  @BeforeEach
+  protected void beforeEach() throws Exception {
+    StorageFactory factory = StorageFactory.create(TestUtils.addSuffix(getProperties(), TEST_NAME));
+    admin = factory.getStorageAdmin();
+    createTable();
+    adminTestUtils = AdminTestUtils.create(TestUtils.addSuffix(getStorageProperties(), TEST_NAME));
+  }
+
+  @AfterEach
+  protected void afterEach() throws Exception {
+    dropTable();
+  }
+
+  @AfterAll
+  protected void afterAll() throws Exception {}
+
+  @Test
+  public void repairTable_ForDeletedMetadataTable_ShouldRepairProperly() throws Exception {
+    // Arrange
+    adminTestUtils.dropMetadataTable();
+
+    // Act
+    admin.repairTable(getNamespace(), getTable(), TABLE_METADATA, getCreateOptions());
+
+    // Assert
+    assertThat(admin.tableExists(getNamespace(), getTable())).isTrue();
+    assertThat(admin.getTableMetadata(getNamespace(), getTable())).isEqualTo(TABLE_METADATA);
+  }
```

Comment:
As a matter of fact, I added the put data verification in `DistributedTransactionAdminRepairTableIntegrationTestBase` because I wanted to verify that the coordinator table metadata were present in the metadata table after the repair. In retrospect, I could have used `storageAdmin.getTableMetadata(coordinatorNamespace, coordinatorTable)` instead which is maybe more appropriate ?

In `DistributedStorageAdminRepairTableIntegrationTestBase`, I thought it was not necessary to add the put data verification as I could check the metadata for the table were present in the metadata table using the admin. If the metadata for the table is present, then the put data verification is superfluous maybe ?

Reply:
Okay, I see. The reason why I thought we could put data in `DistributedStorageAdminRepairTableIntegrationTestBase` is that I thought putting data could check the stored procedure things for Cosmos DB Tables. But on second thought, we already have the tests for the stored procedure things in  `CosmosAdminRepairTableIntegrationTest` and `ConsensusCommitAdminRepairTableIntegrationTestWithCosmos`, so we don't need to put data.

> In retrospect, I could have used storageAdmin.getTableMetadata(coordinatorNamespace, coordinatorTable) instead which is maybe more appropriate?

Maybe yes, using `storageAdmin.getTableMetadata(coordinatorNamespace, coordinatorTable)` is straightforward to me.

Reply:
All right, I will update `ConsensusCommitAdminRepairTableIntegrationTestWithCosmos` to rely on `storageAdmin.getTableMetadata(coordinatorNamespace, coordinatorTable)` and remove the put data.

Reply:
I revised it in https://github.com/scalar-labs/scalardb/pull/613/commits/aa6395429ffd10b52eef7715bc36e0c8dc491a39 and https://github.com/scalar-labs/scalardb/pull/613/commits/c5beaab6ab8aed61f43c9474f107c17120a005aa